File: pedra_papel_tesoura.py
import random

def play(player_choice):
    choices = ['pedra', 'papel', 'tesoura']
    ai_choice = random.choice(choices)
    
    if player_choice == ai_choice:
        return f"Empate! Você escolheu: {player_choice}. IA escolheu: {ai_choice}."
    elif (player_choice == 'pedra' and ai_choice == 'tesoura') or \
         (player_choice == 'papel' and ai_choice == 'pedra') or \
         (player_choice == 'tesoura' and ai_choice == 'papel'):
        return f"Você ganhou! Você escolheu: {player_choice}. IA escolheu: {ai_choice}."
    else:
        return f"IA ganhou! Você escolheu: {player_choice}. IA escolheu: {ai_choice}."

def main():
    player_choice = input("Escolha pedra, papel ou tesoura: ")
    print(play(player_choice))

if __name__ == "__main__":
    main()

File: PedraPapelTesoura.java
import java.util.Random;
import java.util.Scanner;

public class PedraPapelTesoura {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Escolha pedra, papel ou tesoura: ");
        String playerChoice = scanner.nextLine();
        String result = play(playerChoice);
        System.out.println(result);
    }

    public static String play(String playerChoice) {
        String[] choices = {"pedra", "papel", "tesoura"};
        String aiChoice = choices[new Random().nextInt(choices.length)];
        
        if (playerChoice.equals(aiChoice)) {
            return "Empate! Você escolheu: " + playerChoice + ". IA escolheu: " + aiChoice + ".";
        } else if ((playerChoice.equals("pedra") && aiChoice.equals("tesoura")) ||
                   (playerChoice.equals("papel") && aiChoice.equals("pedra")) ||
                   (playerChoice.equals("tesoura") && aiChoice.equals("papel"))) {
            return "Você ganhou! Você escolheu: " + playerChoice + ". IA escolheu: " + aiChoice + ".";
        } else {
            return "IA ganhou! Você escolheu: " + playerChoice + ". IA escolheu: " + aiChoice + ".";
        }
    }
}
